diff --git a/include/fmt/format-inl.h b/include/fmt/format-inl.h
index 945cb912..89e759f8 100644
--- a/include/fmt/format-inl.h
+++ b/include/fmt/format-inl.h
@@ -162,10 +162,13 @@ FMT_API FMT_FUNC auto format_facet<std::locale>::do_put(
 }
 #endif
 
+// NVCC's preprocessor converts 'std::error_code' to `class std::error_code'
+// and then complains that `return class std::error_code' isn't valid!
+using fmt_system_error = std::system_error;
 FMT_FUNC auto vsystem_error(int error_code, string_view fmt, format_args args)
-    -> std::system_error {
+    -> fmt_system_error {
   auto ec = std::error_code(error_code, std::generic_category());
-  return std::system_error(ec, vformat(fmt, args));
+  return fmt_system_error(ec, vformat(fmt, args));
 }
 
 namespace detail {
@@ -1426,7 +1429,7 @@ FMT_FUNC void format_system_error(detail::buffer<char>& out, int error_code,
                                   const char* message) noexcept {
   FMT_TRY {
     auto ec = std::error_code(error_code, std::generic_category());
-    detail::write(appender(out), std::system_error(ec, message).what());
+    detail::write(appender(out), fmt_system_error(ec, message).what());
     return;
   }
   FMT_CATCH(...) {}
